python/tank/deploy/tank_commands/core_upgrade.py

```diff
@@ -15,36 +15,35 @@
 import sys
 import textwrap
 import uuid
-import shutil
-import stat
 import tempfile
-import datetime
 
 from ...util import shotgun
 from ...platform import constants
-from ... import pipelineconfig
 from ... import pipelineconfig_utils
 from ..zipfilehelper import unzip_file
 from .. import util
 
 from . import console_utils
 
-        
-class CoreUpgradeAction(Action):
+
+class CoreUpdateAction(Action):
```

Comment:
this is nitpicky, sorry! But like @robblau I am also wondering if it's worth changing the name. I am reading https://support.apple.com/en-gb/HT201564 and http://english.stackexchange.com/questions/13353/are-there-any-differences-between-update-and-upgrade and to me it feels like Upgrade is actually more appropriate than update. The core command delivers both features and bug fixes, it adds new functionality (e.g. upgrade) as well as making sure a user is kept up to date with the latest fixes (e.g. update).